### DISCUSSION TITLE
StrafeAirMoveType.cpp minimum throttle control

### DISCUSSION THREAD

**Discussion by KyleAnthonyShepherd** [beyond-all-reason/RecoilEngine#785] [4. Feature Requests and Ideas]
As implemented, a StrafeAirMoveType unit chasing and attacking another air unit can only reduce throttle to 30% of its maximum. 
This leads to some fast aircraft unable to stay behind slower aircraft. 
https://github.com/beyond-all-reason/spring/blob/BAR105/rts/Sim/MoveTypes/StrafeAirMoveType.cpp#L785
> const float  angleLim = 1.0f - goalDotFront * 0.7f;

I propose a new UnitDef that allows the 0.7f factor to be defined on a per-unit basis, so that units can disable throttle reduction (value of 0) and make full speed flybys, or even be able to apply reverse throttle (value greater than 1) if rangeLim is also small enough.

**Comment by badosu** [comment]
@KyleAnthonyShepherd What's the status of this discussion, is this still an open issue?

**Reply by KyleAnthonyShepherd** [reply]
Still technically an open issue, but to my knowledge no one using the engine has needed or desired this feature yet (I only raised the idea when trying some experimental BAR air unit tweaks)

**Comment by sprunk** [comment]
As an easy workaround for this specific case I think one can just temporarily reduce the maximum (so the 30% scales down accordingly).

Being arcane with a billion magic constants is a more general problem with fixed-wing (strafing) airplace behaviour, where afaict gamedevs mostly just throw semi-blind number tweaks at a wall and see what sticks enough to make an airplane behave approximately reasonably. I'm not sure how to best approach it but maybe waiting for specific feedback/recommendations from somebody willing to delve into using it would be best.
